Mathlib/Algebra/Lie/Killing.lean

```diff
@@ -210,18 +212,27 @@ lemma lowerCentralSeries_one_inf_center_le_ker_traceForm :
     lowerCentralSeries R L L 1 ⊓ LieAlgebra.center R L ≤ LinearMap.ker (traceForm R L M) := by
   rintro z ⟨hz : z ∈ lowerCentralSeries R L L 1, hzc : z ∈ LieAlgebra.center R L⟩
   ext x
-  suffices ∀ χ : R, traceForm R L (weightSpaceOf M χ x) z x = 0 by
-    simp [traceForm_eq_sum_weightSpaceOf R L M x, this]
-  intro χ
-  replace hz : LinearMap.trace R _ (toEndomorphism R L (weightSpaceOf M χ x) z) = 0 :=
-    trace_toEndomorphism_eq_zero_of_mem_lcs R L _ (le_refl _) hz
-  let f := toEndomorphism R L (weightSpaceOf M χ x) z
-  let g := toEndomorphism R L (weightSpaceOf M χ x) x
-  have h_comm : Commute f g := commute_toEndomorphism_of_mem_center_left _ hzc x
-  have hg : _root_.IsNilpotent (g - algebraMap R _ χ) :=
-    (toEndomorphism R L M x).isNilpotent_restrict_iSup_sub_algebraMap χ
-  rw [traceForm_apply_apply, LinearMap.trace_comp_eq_mul_of_commute_of_isNilpotent χ h_comm hg, hz,
-    mul_zero]
+  rw [traceForm_apply_apply, LinearMap.zero_apply]
```

Comment:
Would you mind commenting this proof a bit?

Reply:
Good idea, done!